This patch backports an upstream fix for the link to the root page of the C
reference. Relevant commits are 855fb71a and 4dddca39.
--- a/Makefile
+++ b/Makefile
@@ -114,6 +114,7 @@
 			 --stringparam chapters-file index-chapters-c.xml	\
 			 --stringparam title "C Standard Library reference"	\
 			 --stringparam name "cppreference-doc-en-c"	\
+			 --stringparam link "c"	\
 			 index2devhelp.xsl index-functions-c.xml > devhelp-index-c.xml
 	xsltproc fix_devhelp-links.xsl devhelp-index-c.xml > cppreference-doc-en-c.devhelp2

@@ -122,6 +123,7 @@
 			 --stringparam chapters-file index-chapters-cpp.xml	\
 			 --stringparam title "C++ Standard Library reference"	\
 			 --stringparam name "cppreference-doc-en-cpp"	\
+			 --stringparam link "cpp"	\
 			 index2devhelp.xsl index-functions-cpp.xml > devhelp-index-cpp.xml
 	xsltproc fix_devhelp-links.xsl devhelp-index-cpp.xml > cppreference-doc-en-cpp.devhelp2

--- a/index2devhelp.xsl
+++ b/index2devhelp.xsl
@@ -26,12 +26,13 @@
 <xsl:param name="chapters-file" select="''"/>
 <xsl:param name="title" select="''"/>
 <xsl:param name="name" select="''"/>
+<xsl:param name="link" select="''"/>

 <xsl:output indent="yes"/>

 <xsl:template match="/index">
   <book title="{$title}" name="{$name}"
-        base="{$book-base}" link="cpp" version="2" language="c++">
+        base="{$book-base}" link="{$link}" version="2" language="c++">
     <xsl:copy-of select="document($chapters-file)"/>
     <functions>
       <xsl:apply-templates mode="process-item" select="child::*"/>
